src/gc/env/gcenv.ee.h

```diff
@@ -78,7 +80,7 @@ class GCToEEInterface
 
     static void GcEnumAllocContexts(enum_alloc_context_func* fn, void* param);
 
-    static void AttachCurrentThread(); // does not acquire thread store lock
+    static bool CreateBackgroundThread(Thread** thread, GCBackgroundThreadFunction threadStart, void* arg);
```

Comment:
On the other hand, the current way ensures that the gh->bgc_thread is set before the gh->bgc_thread_stub is called. Which might be necessary.